packages/core/src/tests/models.test.ts

```diff
@@ -1,54 +1,146 @@
-import { getModel, getEndpoint } from "../models.ts";
+import { getModel, getEndpoint, models } from "../models.ts";
 import { ModelProviderName, ModelClass } from "../types.ts";
 import { describe, test, expect, vi } from "vitest";
 
+// Mock settings
 vi.mock("../settings", () => {
     return {
         default: {
             SMALL_OPENROUTER_MODEL: "mock-small-model",
+            LARGE_OPENROUTER_MODEL: "mock-large-model",
             OPENROUTER_MODEL: "mock-default-model",
             OPENAI_API_KEY: "mock-openai-key",
             ANTHROPIC_API_KEY: "mock-anthropic-key",
             OPENROUTER_API_KEY: "mock-openrouter-key",
+            ETERNALAI_MODEL: "mock-eternal-model",
+            ETERNALAI_URL: "https://mock.eternal.ai",
+            LLAMACLOUD_MODEL_SMALL: "mock-llama-small",
+            LLAMACLOUD_MODEL_LARGE: "mock-llama-large",
+            TOGETHER_MODEL_SMALL: "mock-together-small",
+            TOGETHER_MODEL_LARGE: "mock-together-large",
         },
         loadEnv: vi.fn(),
     };
 });
 
-describe("Model Provider Tests", () => {
-    test("should retrieve the correct model for OpenAI SMALL", () => {
-        const model = getModel(ModelProviderName.OPENAI, ModelClass.SMALL);
-        expect(model).toBe("gpt-4o-mini");
+describe("Model Provider Configuration", () => {
+    describe("OpenAI Provider", () => {
+        test("should have correct endpoint", () => {
+            expect(models[ModelProviderName.OPENAI].endpoint).toBe("https://api.openai.com/v1");
+        });
+
+        test("should have correct model mappings", () => {
+            const openAIModels = models[ModelProviderName.OPENAI].model;
+            expect(openAIModels[ModelClass.SMALL]).toBe("gpt-4o-mini");
```

Comment:
instead of hardcoding we pay want to use the settings and then default to the hard coded values. As this will change for instance when we add o1 and o1 pro support See https://github.com/ai16z/eliza/pull/999/files#diff-c128c91a940e645cc1d7842bec41d065604fce9e1bbc69dfca3f3d089cac52eb

Reply:
Good point